### PR TITLE
Modified the wording of exception messages and created the troubleshooting page

### PR DESCRIPTION
**Pull request by chloe-zh**
*Issue #, if available:*

* [**Issue #320 Exception handling**](https://github.com/opendistro-for-elasticsearch/sql/issues/320)

*Description of changes:*

* Edited some implicit wording of the error messages in our code base
* Added the troubleshooting page in documentation: https://github.com/chloe-zh/sql/blob/existed-exceptions/docs/user/dql/troubleshooting.rst
* TODO: add reference links (reference manual, troubleshooting etc.) to the error messages (would set about this part when the documentation is completed)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: docs/user/dql/troubleshooting.rst [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+.. contents::
+   :local:
+   :depth: 2
+
+
+Narrative
+=========
+
+SQL plugin is stateless for now so mostly the troubleshooting is mainly focused on why a single query fails.
```
Comment (davidcui-amzn) [inline docs/user/dql/troubleshooting.rst]: no need for `mostly` and `mainly` to both be in the sentence, it's redundant.

File: docs/user/dql/troubleshooting.rst [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+**Workaround**
+
+You need to confirm if the syntax is not supported and disable query analysis if that's the case by the following steps:
+
+1. Identify syntax error in failed query, and correct the syntax if the query does not follow MySQL grammar. Go to step 2 if your query is correct in syntax but it still ends up syntax exception.
+
+1. Disable strict query analysis in new ANTLR parser with the following code block.
+
+1. Verify if the query can pass now. If the query fails as well, please create an issue in our `GitHub Issues <https://github.com/opendistro-for-elasticsearch/sql/issues>`_ section to report bugs fixing or request new features.
```
[the remark this answers is not in the document]
Comment (chloe-zh) [reply docs/user/dql/troubleshooting.rst]: I followed the markdown rule by mistake. It's corrected now. Thanks!

File: src/main/java/com/amazon/opendistroforelasticsearch/sql/spatial/WktToGeoJsonConverter.java [hunk 3]
```diff
@@ -29,7 +30,8 @@ public static String toGeoJson(String wkt) {
         wkt = wkt.toLowerCase();
         int startOfCoordinates = wkt.indexOf("(");
         if (startOfCoordinates == -1) {
-            throw new IllegalArgumentException("not valid wkt");
+            throw new IllegalArgumentException(
+                    StringUtils.format("Failed to convert well-know-text [%s] to geometry type", wkt));
```
[the remark this answers is not in the document]
Comment (chloe-zh) [reply src/main/java/com/amazon/opendistroforelasticsearch/sql/spatial/WktToGeoJsonConverter.java]: Done now, thanks!

File: docs/user/dql/troubleshooting.rst (hunk 1 above)
Comment (chloe-zh) [reply docs/user/dql/troubleshooting.rst]: Corrected now, thanks!

File: src/main/java/com/amazon/opendistroforelasticsearch/sql/domain/Condition.java [hunk 4]
```diff
@@ -242,7 +243,7 @@ public Condition(CONN conn, String name, SQLExpr nameExpr, String oper,
         if (OPEAR.operStringToOpear.containsKey(oper)) {
             this.opear = OPEAR.operStringToOpear.get(oper);
         } else {
-            throw new SqlParseException(oper + " is not a supported operation");
+            throw new SqlParseException("Unsupported operation: " + oper);
```
Comment (dai-chen) [inline src/main/java/com/amazon/opendistroforelasticsearch/sql/domain/Condition.java]: Maybe better to use `SemanticAnalysisException` for consistency here and elsewhere.

File: src/main/java/com/amazon/opendistroforelasticsearch/sql/domain/SearchResult.java [hunk 5]
```diff
@@ -135,7 +135,7 @@ private Object covenValue(Aggregation value) throws SqlParseException {
         } else if (value instanceof LongTerms) {
             return value;
         } else {
-            throw new SqlParseException("unknow this agg type " + value.getClass());
+            throw new SqlParseException("Unknown aggregation value type: " + value.getClass());
```
Comment (dai-chen) [inline src/main/java/com/amazon/opendistroforelasticsearch/sql/domain/SearchResult.java]: `value.getClass()` may need to be replaced by `value.getClass().getSimpleName()` for example.

File: src/main/java/com/amazon/opendistroforelasticsearch/sql/parser/ElasticSqlExprParser.java [hunk 6]
```diff
@@ -120,7 +120,7 @@ protected SQLExpr methodRest(SQLExpr expr, boolean acceptLPAREN) {
             return primaryRest(methodInvokeExpr);
         }
 
-        throw new ParserException("not support token:" + lexer.token());
+        throw new ParserException("Syntax error:" + lexer.token());
```
Comment (dai-chen) [inline src/main/java/com/amazon/opendistroforelasticsearch/sql/parser/ElasticSqlExprParser.java]: Same here to throw `SyntaxAnalysisException` for consistency.

File: src/main/java/com/amazon/opendistroforelasticsearch/sql/domain/Condition.java (hunk 4 above)
Comment (chloe-zh) [reply src/main/java/com/amazon/opendistroforelasticsearch/sql/domain/Condition.java]: Done now, thanks!

File: src/main/java/com/amazon/opendistroforelasticsearch/sql/domain/SearchResult.java (hunk 5 above)
Comment (chloe-zh) [reply src/main/java/com/amazon/opendistroforelasticsearch/sql/domain/SearchResult.java]: Done, thanks!

File: src/main/java/com/amazon/opendistroforelasticsearch/sql/parser/ElasticSqlExprParser.java (hunk 6 above)
Comment (chloe-zh) [reply src/main/java/com/amazon/opendistroforelasticsearch/sql/parser/ElasticSqlExprParser.java]: Done, thanks!

**Pull request merged**